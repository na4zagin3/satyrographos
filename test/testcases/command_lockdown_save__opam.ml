module StdList = List

open Satyrographos_testlib
open TestLib

open Shexp_process

let satyristes =
{|
(lang "0.0.3")
(library
  (name "grcnum")
  (version "0.2")
  (sources
    ((package "grcnum.satyh" "./grcnum.satyh")
     (font "grcnum-font.ttf" "./font.ttf" ())
     (hash "fonts.satysfi-hash" "./fonts.satysfi-hash")
     ; (file "doc/grcnum.md" "README.md")
    ))
  (opam "satysfi-grcnum.opam")
  (dependencies (fonts-theano)))

(doc
  (name "example-doc")
  (build
    ((satysfi "doc-example.saty" "-o" "doc-example-ja.pdf")
     (make "build-doc")))
  (dependencies (grcnum fonts-theano)))
|}

let satysfi_grcnum_opam =
  "satysfi-grcnum.opam", TestLib.opam_file_for_test
    ~name:"satysfi-grcnum"
    ~version:"0.1"
    ()

let fontHash =
{|{
  "grcnum:grcnum-font":<"Single":{"src-dist":"grcnum/grcnum-font.ttf"}>
}|}

let makefile =
{|
PHONY: build-doc
build-doc:
	@echo "Target: build-doc"
	@echo 'Files under $$SATYSFI_RUNTIME'
	@echo "=============================="
	@cd "$(SATYSFI_RUNTIME)" ; find . | LC_ALL=C sort
	@echo "=============================="
|}

let files =
  [
    satysfi_grcnum_opam;
    "Satyristes", satyristes;
    "README.md", "@@README.md@@";
    "fonts.satysfi-hash", fontHash;
    "grcnum.satyh", "@@grcnum.satyh@@";
    "font.ttf", "@@font.ttf@@";
    "doc-grcnum.saty", "@@doc-grcnum.saty@@";
    "doc-example.saty", "@@doc-example.saty@@";
    "Makefile", makefile;
  ]

let opam_response = {
  PrepareBin.list_result = {|# Packages matching: (installed | available) & (name-match(satyrographos) | name-match(satysfi) | name-match(ocaml))
# Name       ,# Repository    ,# Version
ocaml        ,default         ,4.09.0
satyrographos,satysfi-external,0.0.2.7
satysfi      ,satyrographos   ,0.0.5+dev2020.09.05|}
}

let env ~dest_dir:_ ~temp_dir : Satyrographos.Environment.t t =
  let open Shexp_process.Infix in
  let pkg_dir = FilePath.concat temp_dir "pkg" in
  let prepare_pkg =
    PrepareDist.empty pkg_dir
    >> prepare_files pkg_dir files
  in
  let empty_dist = FilePath.concat temp_dir "empty_dist" in
  let prepare_dist = PrepareDist.empty empty_dist in
  let opam_reg = FilePath.concat temp_dir "opam_reg" in
  let log_file = exec_log_file_path temp_dir in
  let prepare_opam_reg =
    PrepareOpamReg.(prepare opam_reg theanoFiles)
    >> PrepareOpamReg.(prepare opam_reg grcnumFiles)
    >> PrepareOpamReg.(prepare opam_reg classGreekFiles)
    >> PrepareOpamReg.(prepare opam_reg baseFiles)
  in
  let bin = FilePath.concat temp_dir "bin" in
  prepare_pkg
  >> prepare_dist
  >> prepare_opam_reg
  >> PrepareBin.prepare_bin ~opam_response bin log_file
  >>| read_env ~opam_reg ~dist_library_dir:empty_dist

let () =
  let verbose = false in
  let main env ~dest_dir:_ ~temp_dir ~outf:_ =
    let _name = Some "example-doc" in
    (* let dest_dir = FilePath.concat dest_dir "dest" in *)
    Satyrographos_command.Lockdown.save_lockdown
      ~verbose
      ~env
      ~buildscript_path:(FilePath.concat temp_dir "pkg/Satyristes")
  in
  let post_dump_dirs ~dest_dir:_ ~temp_dir =
    let pkg_dir = FilePath.concat temp_dir "pkg" in
    [pkg_dir]
  in
  eval (test_install ~post_dump_dirs env main)
