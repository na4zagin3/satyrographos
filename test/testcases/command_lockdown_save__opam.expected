Installing packages
------------------------------------------------------------
Gathering OPAM package information...
------------------------------------------------------------
@@temp_dir@@/pkg
@@temp_dir@@/pkg/Makefile
@@temp_dir@@/pkg/README.md
@@temp_dir@@/pkg/Satyristes
@@temp_dir@@/pkg/doc-example.saty
@@temp_dir@@/pkg/doc-grcnum.saty
@@temp_dir@@/pkg/font.ttf
@@temp_dir@@/pkg/fonts.satysfi-hash
@@temp_dir@@/pkg/grcnum.satyh
@@temp_dir@@/pkg/lockdown.yaml
@@temp_dir@@/pkg/satysfi-grcnum.opam
------------------------------------------------------------
diff -Nr @@empty_dir@@/Makefile @@temp_dir@@/pkg/Makefile
0a1,9
> 
> PHONY: build-doc
> build-doc:
> 	@echo "Target: build-doc"
> 	@echo 'Files under $$SATYSFI_RUNTIME'
> 	@echo "=============================="
> 	@cd "$(SATYSFI_RUNTIME)" ; find . | LC_ALL=C sort
> 	@echo "=============================="
> 
diff -Nr @@empty_dir@@/README.md @@temp_dir@@/pkg/README.md
0a1
> @@README.md@@
diff -Nr @@empty_dir@@/Satyristes @@temp_dir@@/pkg/Satyristes
0a1,21
> 
> (lang "0.0.3")
> (library
>   (name "grcnum")
>   (version "0.2")
>   (sources
>     ((package "grcnum.satyh" "./grcnum.satyh")
>      (font "grcnum-font.ttf" "./font.ttf" ())
>      (hash "fonts.satysfi-hash" "./fonts.satysfi-hash")
>      ; (file "doc/grcnum.md" "README.md")
>     ))
>   (opam "satysfi-grcnum.opam")
>   (dependencies (fonts-theano)))
> 
> (doc
>   (name "example-doc")
>   (build
>     ((satysfi "doc-example.saty" "-o" "doc-example-ja.pdf")
>      (make "build-doc")))
>   (dependencies (grcnum fonts-theano)))
> 
diff -Nr @@empty_dir@@/doc-example.saty @@temp_dir@@/pkg/doc-example.saty
0a1
> @@doc-example.saty@@
diff -Nr @@empty_dir@@/doc-grcnum.saty @@temp_dir@@/pkg/doc-grcnum.saty
0a1
> @@doc-grcnum.saty@@
diff -Nr @@empty_dir@@/font.ttf @@temp_dir@@/pkg/font.ttf
0a1
> @@font.ttf@@
diff -Nr @@empty_dir@@/fonts.satysfi-hash @@temp_dir@@/pkg/fonts.satysfi-hash
0a1,3
> {
>   "grcnum:grcnum-font":<"Single":{"src-dist":"grcnum/grcnum-font.ttf"}>
> }
diff -Nr @@empty_dir@@/grcnum.satyh @@temp_dir@@/pkg/grcnum.satyh
0a1
> @@grcnum.satyh@@
diff -Nr @@empty_dir@@/lockdown.yaml @@temp_dir@@/pkg/lockdown.yaml
0a1,16
> satyrographos: 0.0.3
> dependencies:
> - Opam
> - packages:
>   - name: ocaml
>     version: 4.09.0
>   - name: satyrographos
>     version: 0.0.2.7
>   - name: satysfi
>     version: 0.0.5+dev2020.09.05
>   repos:
>   - name: satysfi-external
>     url: git+https://github.com/gfngfn/satysfi-external-repo.git
>   - name: default
>     url: https://opam.ocaml.org/
> autogen: {}
diff -Nr @@empty_dir@@/satysfi-grcnum.opam @@temp_dir@@/pkg/satysfi-grcnum.opam
0a1,30
> opam-version: "2.0"
> synopsis: "Test Package"
> name: "satysfi-grcnum"
> version: "0.1"
> description: """
> Test package for SATySFi
> """
> maintainer: "SAKAMOTO Noriaki <mrty.ityt.pt@gmail.com>"
> authors: "SAKAMOTO Noriaki <mrty.ityt.pt@gmail.com>"
> license: "LGPL-3.0-or-later"
> homepage: "https://github.com/na4zagin3/satysfi-fss"
> dev-repo: "git+https://github.com/na4zagin3/satysfi-fss.git"
> bug-reports: "https://github.com/na4zagin3/satysfi-fss/issues"
> depends: [
> 
>   "satysfi" {>= "0.0.5" & < "0.0.6"}
>   "satyrographos" {>= "0.0.2.6" & < "0.0.3"}
> 
>   "satysfi-base" {>= "1.3.0" & < "2"}
>   "satysfi-fonts-junicode" {>= "1" & < "2"}
> 
> ]
> build: [ ]
> install: [
>   ["satyrographos" "opam" "install"
>    "--name" "test-package"
>    "--prefix" "%{prefix}%"
>    "--script" "%{build}%/Satyristes"]
> ]
> 
------------------------------------------------------------
Command invoked:
opam list -i --color=never --columns name,repository,installed-version --separator=, --recursive --required-by satysfi-fonts-theano,satysfi-grcnum
